crates/win-api-wrappers/src/identity/account.rs

```diff
@@ -213,3 +265,18 @@ impl Drop for ProfileInfo {
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    use crate::str::u16cstr;
+    use crate::token::Token;
+
+    #[test]
+    fn get_virtual_account_name() {
+        let token = Token::current_process_token();
+        let account_name = virtual_account_name(&token).unwrap();
+        assert_eq!(account_name, u16cstr!(""))
+    }
```

Comment:
This fails with
```
The handle is invalid. (0x80070006)
```

Which makes sense given that `Token::current_process_token()` returns a pseudo-handle.

Reply:
Modified into the following code that is working as expected:

```rust
// Retrieve a non-pseudo-token with TOKEN_QUERY | TOKEN_DUPLICATE | TOKEN_IMPERSONATE access rights.
// Without these access rights, Windows will return an "Access is denied." error.
let current_process = Process::current_process();
let token = current_process
    .token(Security::TOKEN_QUERY | Security::TOKEN_DUPLICATE | Security::TOKEN_IMPERSONATE)
    .unwrap();

let account_name = virtual_account_name(&token).unwrap();
let account_name = account_name.as_ucstr().to_string_lossy();

// Check that the UTF-16 substring substition logic is working as expected.
assert!(account_name.contains('_'));
assert!(!account_name.contains('\\'));
```